.github/workflows/ci.yml

```diff
@@ -63,8 +63,58 @@ jobs:
       - name: run integration tests
         run: ./scripts/integration.sh
 
+  gpu-folders:
+    runs-on: [ self-hosted, gpu ]
+    steps:
+      - name: Change dir owner to working user
+        run: sudo chown -R $USER:$USER $GITHUB_WORKSPACE
+      - name: Checkout
+        uses: actions/checkout@v2
+      - id: changed-files
+        uses: jitterbit/get-changed-files@v1
+        continue-on-error: true
+      - id: set-matrix
+        run: bash scripts/get-gpu-executors.sh
+        env:
+          CHANGED_FILES: ${{ steps.changed-files.outputs.all }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  gpu-tests:
+    needs: [ gpu-folders ]
+    runs-on: [ self-hosted, gpu ]
+    strategy:
+      fail-fast: false
+      matrix:
+        folder: ${{fromJson(needs.gpu-folders.outputs.matrix)}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: debug
+        run: echo ${{ matrix.folder }}
+      - name: build docker image
+        run: docker build -t $(echo "$(basename "${{ matrix.folder }}"):gpu" | tr '[:upper:]' '[:lower:]') -f ${{ matrix.folder }}/Dockerfile.gpu ${{ matrix.folder }}
```

Comment:
How about this: we set the default working directory using
```yaml
gpu-tests:
  defaults:
    run:
      working-directory: ${{ matrix.folder }}
```
this way we can use `${PWD##*/}` for container name. And we can also use `-f Dockerfile.gpu` as well. I'm not 100% if it would work, but would look much cleaner if it would